cmd/tkey-sign/main.go

```
@@ -0,0 +1,389 @@
+		pflag.Usage()
+		os.Exit(2)
+	}
+
+	if *versionOnly {
+		le.Printf("tkey-sign %s", version)
+		os.Exit(0)
+	}
+
+	if *helpOnly {
+		pflag.Usage()
+		os.Exit(0)
+
+	}
+
+	if *getKey {
+		cmd = cmdGetKey
+		cmdArgs++
+	}
+
+	if *sign {
+		cmd = cmdSign
+		cmdArgs++
+	}
+
+	if *verify {
+		cmd = cmdVerify
+		cmdArgs++
+	}
+
+	if cmdArgs > 1 {
+		pflag.Usage()
+		os.Exit(1)
+	}
+
+	switch cmd {
+	case cmdGetKey:
+		if *keyFile == "" {
+			le.Printf("Provide public key file with -p FILE")
+			os.Exit(1)
+		}
+
+		signer, pubkey, err := loadSigner(*devPath, *speed, *ussFile, *enterUss)
+		if err != nil {
+			le.Printf("Couldn't load signer: %v", err)
+			os.Exit(1)
+		}
+
+		if *force {
+			err = writeBase64(*keyFile, pubkey, true)
+		} else {
+			err = writeRetry(*keyFile, pubkey)
+		}
+
+		if err != nil {
+			le.Printf("%v", err)
+			signer.Close()
+			os.Exit(1)
+		}
+
+		signer.Close()
+
+	case cmdSign:
+		if *messageFile == "" {
+			le.Printf("Provide message file with -m FILE")
+			os.Exit(1)
+		}
+
+		if *sigFile == "" {
+			*sigFile = *messageFile + ".sig"
+		}
+
+		signer, pubkey, err := loadSigner(*devPath, *speed, *ussFile, *enterUss)
+		if err != nil {
+			le.Printf("Couldn't load signer: %v", err)
+			os.Exit(1)
+		}
+
+		signature, err := signFile(*signer, pubkey, *messageFile)
+		if err != nil {
+			le.Printf("signing failed: %v", err)
+			signer.Close()
+			os.Exit(1)
+		}
+
+		if *force {
+			err = writeBase64(*sigFile, signature, true)
+		} else {
+			err = writeRetry(*sigFile, signature)
+		}
+		if err != nil {
+			le.Printf("Couldn't store signature: %v", err)
+			signer.Close()
+			os.Exit(1)
+		}
+
+		signer.Close()
+
+	case cmdVerify:
+		if *messageFile == "" {
+			le.Printf("Provide message file with -m FILE")
+			os.Exit(1)
+		}
+
+		if *keyFile == "" {
+			le.Printf("Provide public key file path with -p FILE")
+			os.Exit(1)
+		}
+
+		if *sigFile == "" {
+			*sigFile = *messageFile + ".sig"
```

Maybe we should check if that exists, and if not say "Signature file not found, specify with '-x <file>'" or something. Would be nicer than the the error on row 377 today:
"Error verifying: decodeFileInput: could not read Makefile.sig: open Makefile.sig: no such file or directory".
which absolutely tells precisely the issue. Maybe not super important, but would make it a bit nicer with an instruction in the right direction. 

One could use Stat instead of open to accomplish it if one wants to fail fast:
```
_, err := os.Stat(*sigFile)
if err != nil {
  if os.IsNotExist(err) {
    // File or directory does not exist
  } else {
    // Some other error. The file may or may not exist
  }
}
```

---

I added a check checking the error from readSig() to see if it's ErrNotExist and give the useful hint.